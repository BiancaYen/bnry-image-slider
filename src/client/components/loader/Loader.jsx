// Loader

import PropTypes from 'prop-types';
import React from 'react';

// Styles
import { Indicator, Container } from './styles';

// Prop Types
const propTypes = {
    children: PropTypes.node.isRequired,
    isLoading: PropTypes.bool.isRequired
};

const Loader = ({ isLoading, children }) => {
    if (isLoading) {
        return (
            <Container>
                <Indicator>
                    Loading
                </Indicator>
            </Container>
        );
    }
    return <div>{children}</div>;
};

Loader.propTypes = propTypes;

export default Loader;
