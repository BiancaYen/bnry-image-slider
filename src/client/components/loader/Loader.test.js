// Loader Test

import { mount, shallow } from 'enzyme';
import React from 'react';

// Components
import Loader from './Loader';

// Constants
const content = 'Content';

describe('Loader', () => {
    it('renders with all required prop types', () => {
        const component = shallow(
            <Loader isLoading>
                {content}
            </Loader>
        );

        expect(component).toMatchSnapshot();
    });

    it('asserts that content renders correctly depending on prop type', () => {
        const component = mount(
            <Loader isLoading>
                {content}
            </Loader>
        );

        expect(component.find('div div').text()).toEqual('Loading');
    });

    it('asserts that content renders correctly depending on prop type', () => {
        const component = mount(
            <Loader isLoading={false}>
                {content}
            </Loader>
        );

        expect(component.find('div').text()).toEqual(content);
    });
});
