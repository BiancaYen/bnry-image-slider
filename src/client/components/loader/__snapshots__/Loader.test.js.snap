// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders with all required prop types 1`] = `
<Container>
  <Indicator>
    Loading
  </Indicator>
</Container>
`;
